src/main/java/AWSClientFactory.java

```diff
@@ -12,42 +12,49 @@
  */
 
 import com.amazonaws.ClientConfiguration;
+import com.amazonaws.SdkClientException;
 import com.amazonaws.auth.AWSCredentials;
 import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
 import com.amazonaws.services.codebuild.AWSCodeBuildClient;
 import com.amazonaws.services.codebuild.model.InvalidInputException;
 import com.amazonaws.services.logs.AWSLogsClient;
 import com.amazonaws.services.s3.AmazonS3Client;
+import lombok.Getter;
+import org.apache.commons.lang.StringUtils;
+
 
 public class AWSClientFactory {
 
-    private final String proxyHost;
-    private final String proxyPort;
-    private final String awsAccessKey;
-    private final String awsSecretKey;
+    @Getter private final boolean isDefaultCredentialUsed;
     private final String region;
-
     private ClientConfiguration clientConfig;
     private AWSCredentials awsCredentials;
 
     public AWSClientFactory(String proxyHost, String proxyPort, String awsAccessKey, String awsSecretKey, String region) throws InvalidInputException {
 
-        Validation.checkAWSClientFactoryConfig(proxyHost, proxyPort, awsAccessKey, awsSecretKey);
+        // Priority is IAM credential > Credentials provided by the default AWS credentials provider
+        if(StringUtils.isNotEmpty(awsAccessKey) && StringUtils.isNotEmpty(awsSecretKey)) {
+            awsCredentials = new BasicAWSCredentials(awsAccessKey,awsSecretKey);
+            isDefaultCredentialUsed = false;
+        } else {
+            try {
+                awsCredentials = DefaultAWSCredentialsProviderChain.getInstance().getCredentials();
```

Comment:
These credentials expire, so you shouldn't save them into a separate property.  You need to construct the client with provider itself.  Alternatively, you could construct the client without supplying any credentials, and it will use the default provider chain by default:
http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html